src/reactor.cc

```diff
@@ -0,0 +1,375 @@
+#include "reactor.h"
+
+#include "boost/shared_ptr.hpp"
+
+namespace tricycle {
+
+Reactor::Reactor(cyclus::Context* ctx) : cyclus::Facility(ctx) {
+  // capacities set somewhat arbitrarily
+  fuel_tracker.Init({&tritium_storage}, 1000.0);
+  blanket_tracker.Init({&blanket}, 100000.0);
+}
+
+std::string Reactor::str() {
+  return Facility::str();
+}
+
+void Reactor::Tick() {
+  if (sufficient_tritium_for_operation) {
+    SequesterTritium();
+    OperateReactor(TBR);
+    blanket_fill_policy.Start();
+    Reactor::RecordStatus("Online", fusion_power);
+  } else {
+    Reactor::RecordStatus("Shut-down", 0);
+  }
+
+
+  DecayInventory(tritium_storage);
+  DecayInventory(tritium_excess);
+  DecayInventory(tritium_sequestered);
+
+  ExtractHelium(tritium_storage);
+  ExtractHelium(tritium_excess);
+
+  if (!tritium_storage.empty() && sufficient_tritium_for_operation) {
+    double surplus = std::max(
+      tritium_storage.quantity() - reserve_inventory, 0.0);
+
+    cyclus::Material::Ptr storage_fuel = tritium_storage.Pop();
+    tritium_excess.Push(storage_fuel->ExtractQty(surplus));
+
+    if (surplus > 0.0) {
+      RecordOperationalInfo(
+          "Tritium Moved",
+          std::to_string(surplus) + "kg of T moved from storage to excess");
+    }
+    tritium_storage.Push(storage_fuel);
+    CombineInventory(tritium_excess);
+  }
+
+  // This pulls out some of the blanket each timestep so that fresh blanket can
+  // be added.
+  // NOTE: currently spent blanket dissapears into the ether. This needs to be
+  // changed in version 1 for materials tracking.
+
+  double blanket_turnover = blanket_size * blanket_turnover_rate;
+  if (!blanket.empty() && blanket.quantity() >= blanket_turnover && 
+      context()->time() % blanket_turnover_frequency == 0) {
```

Comment:
None of this should happen unless this is true first

Reply:
I'm not entirely sure what this is referring to, I'll be honest...

Reply:
The last part of your conditional tests whether it is time to do blanket turnover.  I think this should be tested first by itself, and then the rest of the tests should only happen if it's true.

Reply:
ohh, gotcha. After looking at it I agree. Change made. Good call. Thanks!